File: Demos/Datasnap/Server/proxy/objectivec_ios42/Proxy/TJSONTrue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TJSONTrue.h"


@implementation TJSONTrue
-(NSString *) asJSONString{
 return @"true";
}
-(id) getInternalObject {
	return [NSNumber numberWithBool:YES];
}
-(NSString *) toString{
	return [self asJSONString];
}
-(JSONValueType) getJSONValueType{
	return JSONTrue;
}

@end

File: Demos/Datasnap/Server/proxy/java_blackberry/com/embarcadero/javablackberry/TJSONTrue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javablackberry;

/**
 * 
 * Contains and returns the boolean value true like a JSON.
 * 
 */

public class TJSONTrue extends TJSONValue {

	/**
	 * Returns the String "true"
	 * 
	 * @return
	 */
	public String asJSONString() {
		return "true";
	}

	public Object getInternalObject() {
		return Boolean.TRUE;
	}

	public int getJsonValueType() {
		return JSONValueType.JSONTrue;
	}

	public String toString() {
		return asJSONString();
	}
}
